pubtools/pulplib/_impl/client/client.py

```diff
@@ -203,6 +203,36 @@ def search_repository(self, criteria=None):
             Repository, "repositories", criteria=criteria, search_options=search_options
         )
 
+    def search_repository_content(self, repository_id, type_ids, criteria=None):
+        """Search the given repository for content matching the given criteria.
+
+        Args:
+            repository_id (str)
+                The ID of the repository in which to search
+            type_ids (str, list, tuple)
+                A list of content types to search
+            criteria (:class:`~pubtools.pulplib.Criteria`)
+                A criteria object used for this search.
+
+        Returns:
+            list[:class:`~pubtools.pulplib.Unit`]
+                A list of zero or more :class:`~pubtools.pulplib.Unit`
+                subclasses found by the search operation.
+
+        Raises:
+            :class:`~pubtools.pulplib.InvalidContentTypeException`
+                If the criteria does not contain a valid _content_type_id.
+
+        .. versionadded:: 2.4.0
+        """
+        search_f = self._search(
+            Unit,
+            "repositories/%s" % repository_id,
+            criteria=criteria,
+            type_ids=validate_type_ids(type_ids),
+        )
+        return [unit for unit in search_f.result().as_iter()]
```

Comment:
.result() and .as_iter() are not required as these are implicit in f_proxy and Page classes. So this can be [unit for unit in search_f]

Reply:
Oh, cool! Thanks!